packages/core/core/src/services/document-service/transform/__tests__/fields.test.ts

```diff
@@ -0,0 +1,34 @@
+import { transformFields } from '../fields';
+
+describe('transformFields', () => {
+  it('should transform a single input', () => {
+    const input = ['id'];
+    const expected = ['documentId'];
+    expect(transformFields(input)).toEqual(expected);
+  });
+
+  it('should transform an array of inputs', () => {
+    const input = ['id', 'name'];
+    const expected = ['documentId', 'name'];
+    expect(transformFields(input)).toEqual(expected);
+  });
+
+  it('should transform multiple inputs', () => {
+    // TODO what if a key is repeated?
+    // Do we handle this?
+    const input = ['id', 'name', 'id'];
+    const expected = ['documentId', 'name', 'documentId'];
+    expect(transformFields(input)).toEqual(expected);
+  });
+
+  it('should not modify other inputs', () => {
+    const input = ['id', 'name', 'description'];
+    const expected = ['documentId', 'name', 'description'];
+    expect(transformFields(input)).toEqual(expected);
+  });
+
+  it('should throw if there is no id present in the fields', () => {
+    const input = ['name', 'description'];
+    expect(() => transformFields(input)).toThrow();
```

Comment:
I believe this should be a valid case, even if the id is not present, the database always selects and returns the `id` field